lib/puppet/provider/openldap_database/olc.rb

```diff
@@ -131,10 +131,29 @@ def exists?
     @property_hash[:ensure] == :present
   end
 
+  def fetch_index
+arg = ['-b',
+      'cn=config',
+      '-H',
+      "ldap:///???(&(objectClass=olc#{@property_hash[:backend].to_s.capitalize}Config)(olcSuffix=#{@property_hash[:suffix]}))"]
+    $stderr.puts(arg.inspect)
+    slapcat(
+      '-b',
+      'cn=config',
+      '-H',
+      "ldap:///???(&(objectClass=olc#{@property_hash[:backend].to_s.capitalize}Config)(olcSuffix=#{@property_hash[:suffix]}))").split("\n").collect do |line|
+      if line =~ /^olcDatabase: /
+        @property_hash[:index] = line.match(/^olcDatabase: \{(\d+)\}#{@property_hash[:backend]}$/).captures[0].to_i
+      end
+    end
+  end
```

Comment:
That would be too much. self.instances creates new objects, this method only looks for the combination of a database config with a specific suffix. The result is only two lines (in theory) and the updated index for this particular suffix.